components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java

```diff
@@ -91,7 +92,11 @@ public boolean process(Exchange exchange, AsyncCallback async) {
         final Boolean await = msg.getHeader(MiloConstants.HEADER_AWAIT, this.defaultAwaitWrites, Boolean.class);
 
         if (TRUE.equals(await)) {
-            future.whenComplete((v, ex) -> async.done(false));
+            future.whenComplete((result, throwable) -> {
+                        msg.setBody(result);
```

Comment:
Ah what if throwable != null, eg I assume there was an error, then its likely better to set response, on the exchange

if (throwable != null) {
msg.getExchange().setException(throwable);
} else {
  msg.setBody(result);
}

Reply:
Ok, I amended the commit where I added that and also removed an unnecessary import also.